pkg/pulumiyaml/run.go

```diff
@@ -1211,6 +1211,11 @@ func (e *programEvaluator) registerConfig(intm configNode) (interface{}, bool) {
 					c.Type.Value, ctypes.ConfigTypes)
 			}
 
+			if t == ctypes.Int && expectedType == ctypes.Number {
```

Comment:
Should we not check that the default value is safely castable to int. Like if I say type: int, default: 4.3 I think I should still get an error.

Reply:
Yeah, you're right. Fixed.